extensions/amp-lightbox-gallery/0.1/amp-lightbox-gallery.js

```diff
@@ -867,11 +867,17 @@ export class AmpLightboxGallery extends AMP.BaseElement {
     const anim = new Animation(this.element);
     let duration = MIN_TRANSITION_DURATION;
 
+    const imageBox = /**@type {?}*/ (this.getCurrentElement_().imageViewer)
+        .implementation_.getImageBoxWithOffset();
+
+    // If the image isn't initialized, bail out the transition.
+    if (imageBox.width == 0 && imageBox.height == 0) {
```

Comment:
Hmm this seems like a proxy for something else like a lifecycle method. Is there a class method or hook you can use to know if the image viewer is initialized?

Reply:
So for this, I refactored `<amp-image-viewer>` to return null in the uninitialized case, so that we do a null check instead. I think doing a fade-in is better than waiting for the load, since the image could still be loading while we're animating the fade.